fix: strengthen upgrade tests and show removed packages in UX (#3630)
Shopify/hydrogen: .changeset/removed-packages-ux.md

---
'@shopify/cli-hydrogen': patch
---

> Show removed packages in the upgrade confirmation prompt and upgrade instructions file so merchants can see which dependencies will be cleaned up during the upgrade.

Reviewer: nit:

```suggestion
Show removed packages in the upgrade confirmation prompt and upgrade instructions file so that it's visible which dependencies will be cleaned up during the upgrade.
```

or something like that
